orca-queue/src/main/kotlin/com/netflix/spinnaker/orca/q/handler/StartTaskHandler.kt

```diff
@@ -39,22 +46,44 @@ class StartTaskHandler(
   override val stageDefinitionBuilderFactory: StageDefinitionBuilderFactory,
   @Qualifier("queueEventPublisher") private val publisher: ApplicationEventPublisher,
   private val taskResolver: TaskResolver,
-  private val clock: Clock
+  private val clock: Clock,
+  private val environment: Environment
 ) : OrcaMessageHandler<StartTask>, ExpressionAware {
 
   override fun handle(message: StartTask) {
     message.withTask { stage, task ->
-      task.status = RUNNING
-      task.startTime = clock.millis()
-      val mergedContextStage = stage.withMergedContext()
-      repository.storeStage(mergedContextStage)
+      if (isTaskSkippable(task) && !isTaskEnabled(task)) {
+        log.debug("Skipping task.type=${task.type} because ${getTaskToggle(task)}=false")
+        task.status = SKIPPED
+        val mergedContextStage = stage.withMergedContext()
+        repository.storeStage(mergedContextStage)
 
-      queue.push(RunTask(message, task.id, task.type))
+        queue.push(CompleteTask(message, SKIPPED))
+        publisher.publishEvent(TaskComplete(this, mergedContextStage, task))
+      } else {
+        task.status = RUNNING
+        task.startTime = clock.millis()
+        val mergedContextStage = stage.withMergedContext()
+        repository.storeStage(mergedContextStage)
 
-      publisher.publishEvent(TaskStarted(this, mergedContextStage, task))
+        queue.push(RunTask(message, task.id, task.type))
+        publisher.publishEvent(TaskStarted(this, mergedContextStage, task))
+      }
     }
   }
 
+  fun getTaskToggle(task: TaskExecution): String {
+    val justTheClassName = task.implementingClass.substring(task.implementingClass.lastIndexOf('.') + 1)
+    val snakifiedClassName = CaseFormat.UPPER_CAMEL.to(CaseFormat.LOWER_HYPHEN, justTheClassName)
+    return "tasks.$snakifiedClassName.enabled"
```

Comment:
Snake case is new, and inconsistent. Is there a reason we're diverging from convention for `snake_case` instead of the usual `kebab-case`?

Reply:
Ah, or like @ajordens mentioned, use the task name?

Reply:
`LOWER_HYPHEN` is kebab case, I misnamed it snake case. We can certainly support just the task name, it just seemed less consistent

Reply:
At least in our current configuration we're not using kebab case for `tasks.*` config. It'd be nice to not use kebab unless absolutely necessary.

Reply:
do you have thoughts on stripping vs keeping the package name?

Reply:
I would continue to use the bean name.

`MonitorKatoTask` -> `monitorKatoTask`

Reply:
Trying to figure out the bean name from the class name seems like a can of worms to me. 

`MonitorKatoTask` becomes `monitorKatoTask` but `MDSomething` stays `MDSomething`, and the name could be overriden via annotations or provided by a method name or some custom naming strategy 😵

Reply:
What I really mean is lower camel case.

Reply:
Standardized on `lowerCamelCase` in 11cd05eda, along with docs and a nicer interface